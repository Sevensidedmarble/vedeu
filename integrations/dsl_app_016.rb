#!/usr/bin/env ruby

# frozen_string_literal: true

require 'bundler/setup'
require 'vedeu'

class DSLApp

  Vedeu.bind(:_initialize_) { Vedeu.trigger(:_refresh_) }

  Vedeu.configure do
    debug!
    log '/tmp/vedeu_views_dsl.log'
    renderers [
                Vedeu::Renderers::Terminal.new(
                  filename:   '/tmp/dsl_app_016.out',
                  write_file: true),
                # Vedeu::Renderers::JSON.new(filename: '/tmp/dsl_app_016.json'),
                # Vedeu::Renderers::HTML.new(filename: '/tmp/dsl_app_016.html'),
                # Vedeu::Renderers::Text.new(filename: '/tmp/dsl_app_016.txt'),
              ]
    run_once!
    standalone!
  end

  load './support/test_interface.rb'

  Vedeu.render do
    view(:test_interface) do
      foreground '#ffffff'
      line do
        left   'v->l->left',   { background: '#aa2200', width: 14 }
        right  'v->l->right',  { background: '#0022ff', width: 15 }
      end
      line do
        centre 'v->l->centre', { background: '#007700', width: 30 }
      end
    end
  end

  def self.actual
    File.read('/tmp/dsl_app_016.out')
  end

  def self.expected
    File.read(File.expand_path('../expected/dsl_app_016.out', __FILE__))
  end

  def self.start(argv = ARGV)
    Vedeu::Launcher.execute!(argv)
  end

end # DSLApp

DSLApp.start

if DSLApp.expected == DSLApp.actual
  puts "#{__FILE__} \e[32mPassed.\e[39m"
  exit 0;
else
  puts "#{__FILE__} \e[31mFailed.\e[39m"
  exit 1;
end
