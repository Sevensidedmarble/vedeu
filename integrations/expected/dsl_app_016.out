[0m[0m[2;2H[39m[49m(0l(B[2;3H(0q(B[2;4H [2;5HT[2;6He[2;7Hs[2;8Ht[2;9H [2;10H(0q(B[2;11H(0q(B[2;12H(0q(B[2;13H(0q(B[2;14H(0q(B[2;15H(0q(B[2;16H(0q(B[2;17H(0q(B[2;18H(0q(B[2;19H(0q(B[2;20H(0q(B[2;21H(0q(B[2;22H(0q(B[2;23H(0q(B[2;24H(0q(B[2;25H(0q(B[2;26H(0q(B[2;27H(0q(B[2;28H(0q(B[2;29H(0q(B[2;30H(0q(B[2;31H(0q(B[2;32H(0k(B[0m[3;2H[39m[49m(0x(B[3;3H[38;5;231m[48;5;124mv[3;4H-[3;5H>[3;6Hl[3;7H-[3;8H>[3;9Hl[3;10He[3;11Hf[3;12Ht[3;13H [3;14H [3;15H [3;16H [3;17H[38;5;231m[48;5;21m [3;18H [3;19H [3;20H [3;21Hv[3;22H-[3;23H>[3;24Hl[3;25H-[3;26H>[3;27Hr[3;28Hi[3;29Hg[3;30Hh[3;31Ht[3;32H[39m[49m(0x(B[0m[4;2H[39m[49m(0x(B[4;3H[38;5;231m[48;5;28m [4;4H [4;5H [4;6H [4;7H [4;8H [4;9H [4;10H [4;11H [4;12Hv[4;13H-[4;14H>[4;15Hl[4;16H-[4;17H>[4;18Hc[4;19He[4;20Hn[4;21Ht[4;22Hr[4;23He[4;24H [4;25H [4;26H [4;27H [4;28H [4;29H [4;30H [4;31H [4;32H[39m[49m(0x(B[0m[5;2H[39m[49m(0x(B[5;3H [5;4H [5;5H [5;6H [5;7H [5;8H [5;9H [5;10H [5;11H [5;12H [5;13H [5;14H [5;15H [5;16H [5;17H [5;18H [5;19H [5;20H [5;21H [5;22H [5;23H [5;24H [5;25H [5;26H [5;27H [5;28H [5;29H [5;30H [5;31H [5;32H(0x(B[0m[6;2H[39m[49m(0m(B[6;3H(0q(B[6;4H(0q(B[6;5H(0q(B[6;6H(0q(B[6;7H(0q(B[6;8H(0q(B[6;9H(0q(B[6;10H(0q(B[6;11H(0q(B[6;12H(0q(B[6;13H(0q(B[6;14H(0q(B[6;15H(0q(B[6;16H(0q(B[6;17H(0q(B[6;18H(0q(B[6;19H(0q(B[6;20H(0q(B[6;21H(0q(B[6;22H(0q(B[6;23H(0q(B[6;24H(0q(B[6;25H(0q(B[6;26H(0q(B[6;27H(0q(B[6;28H(0q(B[6;29H(0q(B[6;30H(0q(B[6;31H(0q(B[6;32H(0j(B[0m[0m[0m[0m[0m[0m[0m[0m[0m[0m[0m[0m[0m[0m[0m[0m[0m[0m[0m[0m[0m[0m[0m[0m[0m[0m[0m[0m